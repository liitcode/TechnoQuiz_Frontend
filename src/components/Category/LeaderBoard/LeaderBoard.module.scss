@import '../../../assets/styles/mixins.scss';
@import '../../../assets/styles/variables.scss';
.leaderboard {
  width: 84%;
  padding: 2%;
  height: 450px;
}

.leaders {
  padding: 3%;
  border-radius: 10px;
  overflow-y: scroll;
  box-shadow: 7px 10px 30px;
  padding: 2%;
  border-radius: 10px;
  max-height: 298px;
}
.leaders :nth-child(even) {
  background: #dcdcdc;
}
.leader {
  height: 40px;
  @include set-flex;
  background-color: $background-secondary-color;
  padding: 8%;
  border-radius: 5px;
  justify-content: space-between;
}
.leader :nth-child(even) {
  background: transparent;
}
.leader__index {
  color: $primary-color;
}
.leader__name {
  text-align: left;

  &__container {
    @include set-flex;
  }
}
.leader__score {
  color: $navbar-hover;
}
.leader__img {
  width: 45px;
  height: 45px;
  border-radius: 50%;
  margin: 0 20px;
}

@media only screen and (min-width: 768px) and (max-width: 1023px) {
  .leaderboard {
    width: 40%;
    align-self: flex-start;
  }
  .leaders {
    overflow-y: unset;
    max-height: unset;
  }
}

@media only screen and (min-width: 1024px) {
  .leaderboard {
    width: 30%;
    align-self: flex-start;
  }
  .leaders {
    overflow-y: unset;
    max-height: unset;
  }
}

.leaderboard_img {
  Img {
    width: 40%;
    height: 60%;
    margin-bottom: 5%;
  }
}

.message {
  font-size: 0.8rem;
  margin-bottom: 16px;
}
