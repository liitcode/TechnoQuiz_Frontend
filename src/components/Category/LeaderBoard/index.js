/* eslint-disable no-unused-vars */
import React, { useEffect } from 'react';
import { useSelector,useDispatch } from 'react-redux';
import { AvatarGenerator } from 'random-avatar-generator';
import styles from './LeaderBoard.module.scss';
import Trophy from '../../../assets/images/trophy.png';
import SkeletonLeaderBoard from '../../UI/Skeletons/SkeletonLeaderboard';
import { fetchLeaderBoard } from '../../../Redux/actions/actionCreators/leaderboard'

function LeaderBoard() {
  const dispatch = useDispatch();
  useEffect(async () => {
    dispatch(fetchLeaderBoard());
  }, []);
  const data = useSelector((state) => state.leaders.leaders);
  return (
    <>
      <div className={styles.leaderboard}>
        <div className={styles.leaderboard_img}>
          <img src={Trophy} alt="Leaderboard Trophy" />
        </div>
        {data.length > 1 && (
          <div className={styles.leaders}>
            {/* <Dropdown /> */}
            {/* <div className={styles.message}>
              *Choose Category to see leaders of respective category
            </div> */}
            {data.map((leader, index) => (
              <div className={styles.leader}>
                <div className={styles.leader__name__container}>
                  <div className={styles.leader__index}>{index + 1}</div>
                  <div className={styles.leader__img}>
                    <img
                      className={styles.leader_img}
                      src={new AvatarGenerator().generateRandomAvatar()}
                      alt="leaders"
                    />
                  </div>
                  <div className={styles.leader__name}>{leader.name}</div>
                </div>
                <div className={styles.leader__score}>{leader.score}</div>
              </div>
            ))}
          </div>
        )}
        {data.length === 0 && <SkeletonLeaderBoard />}
      </div>
    </>
  );
}

export default LeaderBoard;
