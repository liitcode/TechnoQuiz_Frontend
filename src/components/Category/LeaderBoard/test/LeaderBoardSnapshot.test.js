import React from "react";
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import storeConfig from '``../../../Redux/store/configureStore';
import LeaderBoard from "../index";

it('renders the component', () => {
    const tree = renderer.create(<Provider store={storeConfig}>
        <MemoryRouter><LeaderBoard /></MemoryRouter></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
})