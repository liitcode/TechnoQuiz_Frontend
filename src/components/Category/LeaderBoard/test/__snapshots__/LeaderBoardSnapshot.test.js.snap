// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="leaderboard"
>
  <div
    className="leaderboard_img"
  >
    <img
      alt="Leaderboard Trophy"
      src="trophy.png"
    />
  </div>
  <div
    className="skeleton-leaders"
  >
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
    <div
      className="skeleton-leader"
    >
      <div
        className="skeleton-avatar-small"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
      <div
        className="skeleton-text"
      >
        <div
          className="shimmer-wrapper"
        >
          <div
            className="shimmer"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
